java/src/main/c++/jni_base_learner.h

```diff
@@ -26,14 +26,14 @@ T base_predict(
 { T result = 0;
   try
   { if (learn)
-      vwInstance->l->learn(*ex);
+      as_singleline(vwInstance->l)->learn(*ex);
```

Comment:
Why do we need to leak type information?   Couldn't the top-level learn() know the type?

Reply:
Todo: Overload VW::learn to take either multiline or singleline examples and check type agreement dynamically, then replace with a call to VW::learn.

Reply:
Done.